basicsr/archs/__init__.py

```diff
@@ -19,7 +19,11 @@
 def build_network(opt):
     opt = deepcopy(opt)
     network_type = opt.pop('type')
-    net = ARCH_REGISTRY.get(network_type)(**opt)
+    try:
+        net = ARCH_REGISTRY.get(network_type)(**opt)
+    except TypeError:
+        net = ARCH_REGISTRY.get(network_type)(opt)
+
```

Comment:
Discuss: 
I think it is better to list all the parameters in the network arch class  (HiFaceGAN in this case) . The reasons are:
1. Architectures can be standalone. If all the parameters are listed in the network arch class, the readers could easily understand what parameters are required for this arch rather than looking through all the network files.  
2. We can also use **default values** for the listed parameters in the network arch class, so that we may not need to list all the parameters in the option file. 
3. HifaceGAN arch has 8 parameters, whose number is acceptable.